DuckDB.NET.Data/DuckDBDataReader.cs

```diff
@@ -191,17 +191,28 @@ public override string GetString(int ordinal)
 
     public override T GetFieldValue<T>(int ordinal)
     {
-        return vectorReaders[ordinal].ColumnDuckDBType switch
+        var value = vectorReaders[ordinal].ColumnDuckDBType switch
         {
-            DuckDBType.List => (T)vectorReaders[ordinal].GetList(rowsReadFromCurrentChunk - 1, (dynamic)Activator.CreateInstance<T>()),
-            DuckDBType.Enum => vectorReaders[ordinal].GetEnum<T>(rowsReadFromCurrentChunk - 1),
-            _ => (T)vectorReaders[ordinal].GetValue(rowsReadFromCurrentChunk - 1)
+            DuckDBType.List => vectorReaders[ordinal].GetList(rowsReadFromCurrentChunk - 1, Activator.CreateInstance<T?>() as dynamic),
```

Comment:
Why change the direct cast with as?

Reply:
GetList expect a `T?`. You've two options a direct cast but then to `dynamic?` or a `as` to `dynamic`. TBH, I'm not sure that I really get the ins/outs of GetList<T?>, so I can definitively be short on this.

Reply:
GetList if using kind of a (dynamic) magic now that allowed me to clean up lots of code but I'm not sure if it's worth it.